### PR TITLE
[1.2.x] Throw IllegalStateException when undeploying nonexistent app

### LINKED ISSUE #45: Throw IllegalStateException when undeploying nonexistent app
Based on [this](https://github.com/spring-cloud/spring-cloud-deployer/blob/master/spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/app/AppDeployer.java#L127), the AppDeployer implementation should throw `IllegalStateException` when undeploying an app that has not been deployed.

### PR DESCRIPTION
**Pull request by ilayaperumalg**
Resolves #45

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
This PR depends on https://github.com/spring-cloud/spring-cloud-deployer/pull/188

File: src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java [hunk 1]
```diff
@@ -197,6 +197,10 @@ public void undeploy(String id) {
 			try {
 				App app = marathon.getApp(id).getApp();
 				logger.debug(String.format("Deleting application: %s", app.getId()));
+				AppStatus status = status(app.getId());
+				if (status.getState().equals(DeploymentState.unknown)) {
+					throw new IllegalStateException(String.format("App '%s' is not deployed", app.getId()));
```
Comment (trisberg) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java:202]: When merging update message to say "App '%s' is not in a deployed state."

File: src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java [hunk 2]
```diff
@@ -219,6 +223,10 @@ private void deleteAppsForGroupDeployment(String groupId) throws MarathonExcepti
 		Group group = marathon.getGroup(groupId);
 		for (App app : group.getApps()) {
 			logger.debug(String.format("Deleting application %s in group %s", app.getId(), groupId));
+			AppStatus status = status(app.getId());
+			if (status.getState().equals(DeploymentState.unknown)) {
+				throw new IllegalStateException(String.format("App '%s' is not deployed", app.getId()));
```
Comment (trisberg) [inline src/main/java/org/springframework/cloud/deployer/spi/mesos/marathon/MarathonAppDeployer.java:228]: When merging update message to say "App '%s' is not in a deployed state."

**Comment by trisberg** [comment]
Don't merge until master is on 1.2.0.BUILD-SNAPSHOT

**Pull request closed without merging**